### PR TITLE
clippy and some chores

### PR DESCRIPTION
**Pull request by drahnr**
mozilla#718

### CODE REVIEW

**Comment by drahnr** [comment]
The <btn>update</btn> does a merge commit of `master`, not a `rebase`.. sigh

**Review by kirushik — APPROVED** [review]
I'm not familiar with the codebase _at all_ (so the depth of my review is severely limited), but nothing in the PR seemed as off to me.

**Review by gww-parity — APPROVED** [review]
LGTM. Although I would appreciate somebody DevOps to double check .gitlab-ci.yml after me.

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -71,11 +71,10 @@ nightly-test:
   <<:                              *docker-env
   stage:                           test
   variables:
-    EXTRA_FEATURES:                "$EXTRA_FEATURES unstable"
+    EXTRA_FEATURES:                "unstable"
   script:
-    - cargo build --verbose --features="all ${EXTRA_FEATURES}" || exit 1
-    - RUST_BACKTRACE=1 cargo test --workspace --verbose --no-default-features --features="${EXTRA_FEATURES}"
-    - RUST_BACKTRACE=1 cargo test --workspace --verbose --features="all ${EXTRA_FEATURES}"
+    - cargo +nightly build --verbose --features="${EXTRA_FEATURES}" || exit 1
```
Comment (TriplEight) [inline .gitlab-ci.yml:76]: nightly is the default, not necessary to `+nightly`

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -84,7 +83,7 @@ stable-test:
   before_script:
     - mkdir -p ./artifacts/sccache/
   script:
-    - cargo +stable build --verbose --all-features
-    - RUST_BACKTRACE=1 cargo +stable test --workspace --verbose --no-default-features
-    - RUST_BACKTRACE=1 cargo +stable test --workspace --verbose --all-features
+    - cargo +stable build --verbose
+    - RUST_BACKTRACE=1 cargo +stable test --workspace --verbose
+    - cargo +stable build --release --features="dist-client,dist-server"
```
Comment (TriplEight) [inline .gitlab-ci.yml:88]: This build job will barely reuse the cache from the previous one due to the different cargo profile. And will leave not reusable cache after itself, so the next job will take more time.

**Review by TriplEight — CHANGES_REQUESTED** [review]

File: .gitlab-ci.yml (hunk 1 above)
Comment (drahnr) [reply .gitlab-ci.yml:76]: I prefer t be explicit rather than implicit.

File: .gitlab-ci.yml (hunk 2 above)
Comment (drahnr) [reply .gitlab-ci.yml:88]: How so? It's debug vs release anyways, so there is nothing shared.

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -84,7 +83,7 @@ stable-test:
   before_script:
     - mkdir -p ./artifacts/sccache/
   script:
-    - cargo +stable build --verbose --all-features
-    - RUST_BACKTRACE=1 cargo +stable test --workspace --verbose --no-default-features
-    - RUST_BACKTRACE=1 cargo +stable test --workspace --verbose --all-features
+    - cargo +stable build --verbose
```
Comment (TriplEight) [inline .gitlab-ci.yml:86]: why is this needed?

File: .gitlab-ci.yml (hunk 2 above)
Comment (TriplEight) [reply .gitlab-ci.yml:88]: Checked, it doesn't reuse even 20% of the previous job cache. Let's put it to another CI job.

Comment (TriplEight) [reply .gitlab-ci.yml:88]: artifacts are orthogonal, exactly, so there's not much sense in having this job here. It rather has to be as a prep job for the `dist-test` job

Comment (TriplEight) [reply .gitlab-ci.yml:88]: which is not yet designed

File: .gitlab-ci.yml (hunk 3 above)
Comment (drahnr) [reply .gitlab-ci.yml:86]: Because the integration tests and some unit tests require the compiled debug binary of sccache.

File: .gitlab-ci.yml (hunk 2 above)
Comment (drahnr) [reply .gitlab-ci.yml:88]: Cache usage is not much of concern of this PR, just to get the pipeline green so we can finally enforce it.

**Review by TriplEight — APPROVED** [review]
This is easy, LGTM
